### PR TITLE
Add RV128I opcodes

### PR DESCRIPTION
**Pull request by pavelkryukov**
With new encoding scheme, 128-bit opcodes are less intrusive.
I'm OK to merge this PR to rv128 branch which is stale now.

### CODE REVIEW

**Comment by pavelkryukov** [comment]
done

**Comment by neelgala** [comment]
@pavelkryukov I don't see the changes on this PR !

**Comment by aswaterman** [comment]
RV128 belongs in the `unratified` directory.

**Comment by aswaterman** [comment]
Everything related to RV128 is unratified, so yes.

I’ll let @neelgala comment as to how to accomplish this. One route is to make the RV128 SLLI etc. pseudo, then make the RV64 SLLI etc. non-pseudo.

**Comment by neelgala** [comment]
its okay to have ratified ops be pseudo ops of unratified ops (example [zext.h](https://github.com/riscv/riscv-opcodes/blob/master/rv32_zbb#L1)). The structure and parse.py script allows you to have this and build artifacts for exactly the extensions you want (only ratified, or a mix of ratified and unratified)

File: rv64_i [hunk 1]
```diff
@@ -3,13 +3,17 @@
...
+$pseudo_op rv128_i::slli slli   rd rs1 31..26=0  shamtd 14..12=1 6..2=0x06 1..0=3
+$pseudo_op rv128_i::srli srli   rd rs1 31..26=0  shamtd 14..12=5 6..2=0x06 1..0=3
+$pseudo_op rv128_i::srai srai   rd rs1 31..26=16 shamtd 14..12=5 6..2=0x06 1..0=3
+
 addiw   rd rs1 imm12            14..12=0 6..2=0x06 1..0=3
-slliw   rd rs1 31..25=0  shamtw 14..12=1 6..2=0x06 1..0=3
-srliw   rd rs1 31..25=0  shamtw 14..12=5 6..2=0x06 1..0=3
-sraiw   rd rs1 31..25=32 shamtw 14..12=5 6..2=0x06 1..0=3
+
+$pseudo_op rv128_i::slli slliw   rd rs1 31..25=0  shamtw 14..12=1 6..2=0x06 1..0=3
```
[the remark this answers is not in the document]
Comment (pavelkryukov) [reply rv64_i]: fixed

File: rv64_i [hunk 2]
```diff
@@ -3,13 +3,17 @@
 lwu     rd rs1       imm12 14..12=6 6..2=0x00 1..0=3
 ld      rd rs1       imm12 14..12=3 6..2=0x00 1..0=3
 sd     imm12hi rs1 rs2 imm12lo 14..12=3 6..2=0x08 1..0=3
-slli    rd rs1 31..26=0  shamt 14..12=1 6..2=0x04 1..0=3
-srli    rd rs1 31..26=0  shamt 14..12=5 6..2=0x04 1..0=3
-srai    rd rs1 31..26=16 shamt 14..12=5 6..2=0x04 1..0=3
+
+$pseudo_op rv128_i::slli slli   rd rs1 31..26=0  shamtd 14..12=1 6..2=0x06 1..0=3
```
[the remark this answers is not in the document]
Comment (pavelkryukov) [reply rv64_i]: fixed

**Review by neelgala — APPROVED** [review]

**Comment by aswaterman** [comment]
Thank you, @pavelkryukov and @neelgala!

**Comment by pavelkryukov** [comment]
Thanks! I guess you can remove this branch now: https://github.com/riscv/riscv-opcodes/tree/rv128

**Pull request merged**